docs/getting-started/architecture.mdx

```diff
@@ -5,59 +5,96 @@ sidebar_position: 2
-      functionality for a specific resource (e.g. a database).
-    * **Processor** - processors are stateless components that operate on a single record and can execute arbitrary
-      actions before forwarding the record to the next node in the pipeline. A processor can also choose to drop a
-      record without forwarding it. They can be attached either to a connector or to a pipeline, based on that they are
-      either processing only records that flow from/to a connector or all records that flow through a pipeline.
-      * **Processor Plugin** - interfaces with Conduit on one side, and with the standalone processor plugin on the
-      other and facilitates the communication between them. A standalone processor plugin is a WASM binary that
-      implements the interface defined in
-      [conduit-processor-sdk](https://github.com/ConduitIO/conduit-processor-sdk) and provides the logic for processing
-      a record (e.g. transforming its content).
-  * **Pipeline Engine** - the pipeline engine consists of nodes that can be connected together with Go channels to form
-    a data pipeline.
-    * **Node** - a node is a lightweight component that runs in its own goroutine and runs as long as the incoming channel
-      is open. As soon as the previous node stops forwarding records and closes its out channel, the current node also
-      stops running and closes its out channel. This continues down the pipeline until all records are drained and the
-      pipeline gracefully stops. In case a node experiences an error all other nodes will be notified and stop running
-      as soon as possible without draining the pipeline.
-* **Persistence** - this layer is used directly by the orchestration layer and indirectly by the core layer (through
-  stores) to persist data. It provides the functionality of creating transactions and storing, retrieving and deleting
-  arbitrary data like configurations or state.
+## API layer
+
+Exposes the public APIs used to communicate with Conduit. It exposes 2 types of APIs:
+
+### gRPC
+
+This is the main API provided by Conduit. The gRPC API definition can be found in [api.proto](https://github.com/ConduitIO/conduit/blob/main/proto/api/v1/api.proto), it can be used to generate code for the client.
+
+### HTTP
+
+The HTTP API is generated using [grpc-gateway](https://github.com/grpc-ecosystem/grpc-gateway) and forwards the requests to the gRPC API. Conduit exposes an openapi definition that describes the HTTP API, which is also exposed through Swagger UI on `http://localhost:8080/openapi/`.
+
+## Orchestration layer
+
+The orchestration layer is responsible for coordinating the flow of operations between the core services. It also takes care of transactions, making sure that changes made to specific entities are not visible to the outside until the whole operation succeeded. There are 3 orchestrators, each responsible for actions related to one of the 3 main entities - pipelines, connectors and processors.
```

Comment:
There are 2 more orchestrators: one for connector plugins and one for processor plugins.

Reply:
Great catch

Reply:
No need to change anything, but technically it's all a single orchestrator, the code is just organized into those 5 types, but behind them is the same orchestrator pointer.